ion/ion.c

```diff
@@ -21,7 +34,7 @@ void *xrealloc(void *ptr, size_t num_bytes) {
 }
 
 void *xmalloc(size_t num_bytes) {
-    void *ptr = malloc(num_bytes);
+    void *ptr = MALLOC(num_bytes, size_t);
```

Comment:
This is not the same as in the line above!
old: malloc num_bytes
new: malloc num_bytes * sizeof(size_t)  ==> malloc num_bytes * 4 (on most 32 bit architectures).

Reply:
Yeah you are write, I suppose you could do MALLOC(num_bytes, char).

Reply:
better:
`#define MALLOC(size, type) ((type*) malloc((size)))
`
or rename `num_bytes` to `nr_elements`

Reply:
I would choose the first. So if would want to allocate 5 ints you should pass 5 * sizeof(int). I'll change it.